### PR TITLE
feat: signal update success to api

### PR DESCRIPTION
**Pull request by m90**
Ticket https://phabricator.wikimedia.org/T333163

Requires https://github.com/wbstack/api/pull/679

### CODE REVIEW

File: src/main/java/org/wikidata/query/rdf/tool/WbStackUpdate.java [hunk 1]
```diff
@@ -93,6 +97,8 @@ private WbStackUpdate() {
 
     private static void setValuesFromEnvOrDie() {
         if (System.getenv("WBSTACK_API_ENDPOINT") == null
```
Comment (m90) [inline src/main/java/org/wikidata/query/rdf/tool/WbStackUpdate.java:99]: This is now badly named. Should it be `WBSTACK_API_ENDPOINT_GET_BATCHES`?

File: src/main/java/org/wikidata/query/rdf/tool/WbStackUpdate.java [hunk 2]
```diff
@@ -193,18 +201,45 @@ private static JsonArray getBatchesFromApi() throws IOException {
         }
     }
 
+    private static void updateRemoteBatchStatus(int batchId, boolean success) throws IOException {
+        URL obj = new URL(success ? wbStackApiEndpointMarkDone : wbStackApiEndpointMarkFailed);
```
Comment (m90) [inline src/main/java/org/wikidata/query/rdf/tool/WbStackUpdate.java:205]: This is copied from the internet without in-depth understanding what's going on. It seems to work as intended, but I am not entirely sure if this leaks connection handles or similar. Do we have a knowledgeable source at hand who could double-check this?

File: src/main/java/org/wikidata/query/rdf/tool/WbStackUpdate.java (hunk 1 above)
Comment (deer-wmde) [reply src/main/java/org/wikidata/query/rdf/tool/WbStackUpdate.java:99]: I'm fine with renaming it - I guess another approach would be to just pass the API URL and maybe hardcode the endpoint routes (or pass them as well) but I have no preference - actually after typing this I've now convinced myself that I'm in favor of just renaming it

File: src/main/java/org/wikidata/query/rdf/tool/WbStackUpdate.java (hunk 2 above)
Comment (deer-wmde) [reply src/main/java/org/wikidata/query/rdf/tool/WbStackUpdate.java:205]: [This blog post](https://www.codejava.net/java-se/networking/how-to-use-java-urlconnection-and-httpurlconnection)  claims that closing the output stream is enough, but I also found [these approaches](https://stackoverflow.com/questions/9150200/closing-urlconnection-and-inputstream-correctly):
- a) wrapping the connection handling in `try {} catch {} finally {}`, where the connections are closed in the `finally` block 
- b) creating the OutputStream in `try()` like `try(OutputStream os = con.getOutputStream()) {`

File: Dockerfile [hunk 3]
```diff
@@ -1,8 +1,11 @@
 FROM maven:3.6.3-jdk-8 as jarjar
 
-COPY ./ /tmp
 WORKDIR /tmp
-RUN mvn clean compile assembly:single
+COPY ./pom.xml /tmp/pom.xml
```
[the remark this answers is not in the document]
Comment (deer-wmde) [reply Dockerfile:4]: Not that I have any experience with maven but this seems like a great improvement where I can't think of any added risk

**Review by deer-wmde — APPROVED** [review]

**Pull request merged**